analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -202,19 +203,40 @@ private void addOrUpdateActivity(UserAnonymizedEntityHolder userAnonymizedHolder
 	private void addOrUpdateDayTruncatedActivity(UserAnonymizedEntityHolder userAnonymizedHolder, ActivityPayload payload,
 			GoalDto matchingGoal)
 	{
-		ActivityDto lastRegisteredActivity = getLastRegisteredActivity(payload, matchingGoal);
+		Optional<ActivityDto> lastRegisteredActivity = getLastRegisteredActivity(payload, matchingGoal);
+		if (precedesLastRegisteredActivity(payload, lastRegisteredActivity))
+		{
+			Optional<Activity> overlappingActivity = findOverlappingActivity(payload, matchingGoal);
+			if (overlappingActivity.isPresent())
```

Comment:
Wouldn't it be good to add some warning logging that duplicate activity was logged for the same app? Both here and also in case it overlaps with the last registered activity.

Reply:
Later perhaps, but as long as users run the app version that sends the same data up to 750,000 times (that happened to me, seriously), we shouldn't fill the log for this.

Reply:
OK agreed.